### ISSUE TITLE
Play Framework Slick Evolutions Apparently Not Working with CRDB

### ISSUE THREAD

**Issue by patwhite** [cockroachdb/cockroach#16044]
**BUG REPORT**
Trying to use Play Framework's Slick integration with evolutions against cockroach. Works when connection string is switched to point to postgres. Appears to actually cause play to crash.

I setup a sample project that demonstrates the problem.

1. Clone https://github.com/patwhite/play-scala-cockroach
2. With Scala and SBT installed, in the root of the project run: sbt run
3. Try clicking the "Apply this script now" button

It crashes, but appears to just drop you on the same page. There will be a table created "play_evolutions", but the users table will not be created, and as near as I can tell no relevant logs obviously show up in crdb's logs or play's.

Logs:
CockroachDB node starting at 2017-05-20 16:27:45.854528319 +0000 UTC
build:      CCL v1.0 @ 2017/05/10 13:19:57 (go1.8.1)
admin:      http://roach1:8080
sql:        postgresql://root@roach1:26257?sslmode=disable
logs:       /cockroach/cockroach-data/logs
store[0]:   path=/cockroach/cockroach-data
status:     initialized new cluster
clusterID:  cd6a74ab-576f-476d-9c30-dc4c4154a5fa
nodeID:     1

**Comment by jordanlewis** [comment]
Thanks for the report and sorry that you're experiencing this difficulty. As far as I know, no one has tried using the Play framework with CockroachDB yet - you may be the first one!

If you have some time, could you try starting CockroachDB with the additional flag `--vmodule=executor=2` and attempting to run the script again? That will emit all of the SQL statements being executed to the logs, and will likely make the root cause of the issue apparent. If you attach those logs to this issue we'll be able to track down what's going on.

**Comment by patwhite** [comment]
Definitely, and I know it generally works with slick, I've used that before, but the play evolutions are the issue.

Verbose logs here: https://gist.github.com/patwhite/c9066ce14209ab6ef51c683c07dd5399

**Comment by patwhite** [comment]
This might be related, I saw something similar around MSSQL, related to how play is using timestamp: https://github.com/playframework/play-slick/issues/33

**Comment by jordanlewis** [comment]
It's not obvious to me what's going wrong here. The main thing that stands out to me is that the `INSERT INTO play_evolutions` statement is only prepared, not executed: https://gist.github.com/patwhite/c9066ce14209ab6ef51c683c07dd5399#file-gistfile1-txt-L307-L309 - my guess is what happened here is that something that CockroachDB returned in the pgwire metadata after the prepare phase was unexpected or incorrect, causing Play to fail to execute the statement. I'm not sure why it didn't log anything though.

Based on a quick source dive into the evolutions code, it [looks like this](https://github.com/playframework/playframework/blob/master/framework/src/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala#L160) is where the `INSERT` happens, via [the `prepareAndExecute` helper](https://github.com/playframework/playframework/blob/master/framework/src/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala#L320-L328). I bet adding some exception logging around one of those areas would reveal what's going wrong.

**Comment by jordanlewis** [comment]
Your hunch about timestamps was correct - it was indeed a timestamp parsing issue that was a little bit difficult to detect because of some missing logging. The root cause was the accidental introduction of multiple ways to parse a timestamp value - the one that's used during execution of prepared statements via pgwire was missing the capability of parsing timestamps in the format that the app was sending. Really, these parsers should be unified to prevent further issues of this nature.

I opened #16046 and #16047 to deal with these issues. I have a patch coming to fix the timestamp parsing issue, and I confirmed that it allows your sample app to progress past the migration.
